guides/developer/getting-set-up.md

```diff
@@ -0,0 +1,86 @@
+---
+layout: single
+title: "Getting Set Up for Spinnaker Development"
+sidebar:
+  nav: guides
+---
+
+{% include toc %}
+
+This page describes the steps a Developer should take to fetch Spinnaker's codebase
+and get set up to work on it.
+
+# System Requirements
+
+This guide assumes you have access to a machine with a minimum specification of:
+
+- At least 16 GB of RAM, ideally more
```

Comment:
Haha yes to be honest this is a hedge.  I booted up a VM with 15 GB of RAM and launching a LocalGit quickly gobbled it all up and I ran into memory errors attempting to run Gradle commands.  Bumping the VM up to 26 GB things now run a lot more smoothly.  I'll play with this a bit and try to get a more precise minimum.

Reply:
:+1:

Reply:
Perhaps I'm coming from an odd environment, but I can't remember the last time I've needed anywhere close to 16GB to develop anything in Spinnaker. Is one of the assumptions of this guide that someone would be running _all_ Spinnaker services as part of their development process or something?

Reply:
@robzienert Indeed this is how I run Spinnaker for development.  I rely on `hal deply apply` to launch the whole thing and then target specific ones to restart after code changes with `hal deploy apply --service-name clouddriver`.

Since this is a shortest-path guide and `hal deploy apply` is quick and easy I'm tempted towards sticking with it but I could be convinced otherwise. We could add a call-out here that running a subset of services can drastically reduce resource consumption?

@dorbin @ezimanyi Re: GB of RAM, I restarted my test VM with 16 and I was then able to launch everything and run `./gradlew test` in clouddriver.  Tests completed in 6 minutes or so.  I then restarted again with 18 GB, ran `./gradlew cleanTest && ./gradlew test` and it took about 4 minutes.  18 GB feels like a good-enough floor here so I'll remove the "At least, ideally more" verbiage.

Reply:
@sbwsg Cool, thanks for the overview. I wasn't aware that people did development this way. Would love to followup out-of-band about your setup a bit more - or when we're in your neighborhood in a couple weeks. :)

Reply:
@robzienert I'd definitely be interested in hearing about other ways of doing development.  (I do more or less what @sbwsg described above.)